aixplain/modules/model/__init__.py

```diff
@@ -240,16 +240,25 @@ def run_async(self, data: Union[Text, Dict], name: Text = "model_process", param
         resp = None
         try:
             resp = r.json()
+            if r.status_code == 401:
+                error_msg = "Unauthorized API key"
+            elif 460 <= r.status_code < 470:
+                error_msg = "Subscription-related error"
+            elif 470 <= r.status_code < 480:
+                error_msg = "Billing-related error"
+            elif 480 <= r.status_code < 490:
+                error_msg = "Supplier-related error"
+            elif 490 <= r.status_code < 500:
+                error_msg = "Validation-related error"
+            else:
+                status_code = str(r.status_code)
+                error_msg = f"Status {status_code}: Unspecified error"
             logging.info(f"Result of request for {name} - {r.status_code} - {resp}")
-
             poll_url = resp["data"]
             response = {"status": "IN_PROGRESS", "url": poll_url}
```

Comment:
I'm curious about why it won't go to the except anyway, I mean if we put these conditions before the r.json, it will run into an error when trying to extract the info from json and go to the catch and set the status to fail along with the error message. Please let me know if I'm misunderstanding it.

Reply:
we may just extract the json if none of these errors occur. Otherwise the response should be:

```python
response = { 'status': 'FAILED', "error": error_msg }
```